### PR TITLE
Update Koin to 4.0.1 and refactor configuration

### PR DESCRIPTION
**Pull request by ksharma-xyz**
### TL;DR
Updated Koin DI configuration and upgraded to stable Koin 4.0.1 release

### What changed?
- Refactored Koin configuration to use lambda-based setup instead of `koinConfiguration`
- Upgraded Koin from 4.0.1-Beta1 to stable 4.0.1 release

### How to test?
1. Run the app on both Android and iOS platforms
2. Verify that dependency injection works correctly
3. Check that the navigation and UI components function as expected
4. Confirm that the UserViewModel and UserRepository are properly injected

| Android | iOS |
|--------|--------|
| <video src="https://github.com/user-attachments/assets/ce4dbe8c-1a07-4f9a-9011-5c85e53a3960">  |  <video src="https://github.com/user-attachments/assets/5762934d-1f75-4651-9626-8bb761ec2bc7">  |

### CODE REVIEW

**Comment by ksharma-xyz** [comment]
cc @arnaudgiuliani - Unable to ask for reviews on PR. Do you accept contributions on getting started project ?

**Comment by arnaudgiuliani** [comment]
You can ping me or the team on slack 👍 Need to double check how you can ask review

File: ComposeMultiplatform/composeApp/src/androidMain/kotlin/org/koin/sample/di/AppModule.android.kt [hunk 1]
```diff
@@ -2,10 +2,10 @@ package org.koin.sample.di
 
 import org.koin.android.ext.koin.androidContext
 import org.koin.android.ext.koin.androidLogger
-import org.koin.dsl.koinConfiguration
+import org.koin.core.KoinApplication
 import org.koin.sample.MainApplication
 
-actual fun nativeConfig() = koinConfiguration {
+actual fun nativeConfig(): KoinApplication.() -> Unit = {
```
Comment (arnaudgiuliani) [inline ComposeMultiplatform/composeApp/src/androidMain/kotlin/org/koin/sample/di/AppModule.android.kt:8]: you can keep `koinConfiguration`, easier to understand here.

**Comment by arnaudgiuliani** [comment]
Let me check on my side, about refreshing the example

**Comment by arnaudgiuliani** [comment]
Code has been updated. Let's close it here. Check again and reopen a new one if you want to update 🙏
